plugins/hls-code-range-plugin/src/Ide/Plugin/CodeRange.hs

```diff
@@ -126,6 +148,42 @@ findPosition pos root = go Nothing root
             startOfRight <- _start . _codeRange_range <$> V.headM right
             if pos < startOfRight then binarySearchPos left else binarySearchPos right
 
+findFoldingRanges :: CodeRange -> [Maybe FoldingRange]
+findFoldingRanges root = do
+    let acc = []
+    let node = _codeRange_children root
+    binarySearchFoldingRange node acc
+
+binarySearchFoldingRange :: Vector CodeRange -> [Maybe FoldingRange] -> [Maybe FoldingRange]
+binarySearchFoldingRange v acc
+    | V.null v = []
+    | V.length v == 1 = do
+        headOfCodeRange <- V.headM v
+        let foldingRangesOfRange = createFoldingRange headOfCodeRange
+        let acc' = acc ++ [foldingRangesOfRange]
+        acc'
+    | otherwise = do
+        let (left, right) = V.splitAt (V.length v `div` 2) v
+        _ <- binarySearchFoldingRange left acc
+        binarySearchFoldingRange right acc
+
+createFoldingRange :: CodeRange -> Maybe FoldingRange
+createFoldingRange node1 = do
+    let range = _codeRange_range node1
+    -- let children = _codeRange_children node1
+    let Range startPos endPos = range
+    let Position lineStart _= startPos
+    let Position lineEnd _ = endPos
+    let codeRangeKind = _codeRange_kind node1
+
+    if codeRangeKind == CodeKindComment
+        then Just (FoldingRange lineStart Nothing lineEnd Nothing (Just FoldingRangeComment))
```

Comment:
I'm pretty sure you do want to include the start and end characters here, you should have them.

Reply:
Start and end characters are optional, and it takes the start and end character of the line by default.

I wanted to try if it makes a difference. I'll check it.